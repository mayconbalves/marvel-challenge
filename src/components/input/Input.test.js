import { render, screen, fireEvent } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import Input from './'

describe('<Input />', () => {
  let props = {
    onKeyDown: () => {},
    onChange: () => {},
    placeholder: 'Procure seu heróis',
    value: 'hulk',
    width: '80%',
    type: 'text',
    name: 'hero'
  }

  it('should render correct name element', () => {
    const { container } = render(
      <MemoryRouter>
        <Input {...props} />
      </MemoryRouter>
    )

    expect(
      screen.getByRole('textbox', { Name: 'hero' })
    ).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render correct input placeholder', () => {
    const container = render(
      <MemoryRouter>
        <Input {...props} />
      </MemoryRouter>
    )

    expect(
      container.getByPlaceholderText(/Procure seu heróis/i)
    ).toBeInTheDocument()
  })

  it('shoudl call onChange', () => {
    const setup = () => {
      const utils = render(
        <MemoryRouter>
          <Input {...props} />
        </MemoryRouter>
      )

    const input = utils.getByRole('textbox', { Name: 'hero' })
      return {
        input,
        ...utils
      }
    }

    const { input } = setup()
    fireEvent.change(input, { target: { value: '' } })
    expect(input.value).toBe('hulk')
  })
})
