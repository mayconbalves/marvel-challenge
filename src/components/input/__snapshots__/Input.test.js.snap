// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render correct name element 1`] = `
<input
  class="sc-bdfBwQ hFmnHx"
  name="hero"
  placeholder="Procure seu herÃ³is"
  type="text"
  value="hulk"
  width="80%"
/>
`;
